datafusion/proto/src/physical_plan/to_proto.rs

```diff
@@ -210,6 +211,7 @@ pub fn serialize_physical_expr(
     value: &Arc<dyn PhysicalExpr>,
     codec: &dyn PhysicalExtensionCodec,
 ) -> Result<protobuf::PhysicalExprNode> {
+    let value = snapshot_physical_expr(Arc::clone(value))?;
```

Comment:
I think we should add some comments here about why this method is being called. Perhaps like 

```suggestion
    // Snapshot the expr in case it has dynamic predicate state so
    // it can be serialized
    let value = snapshot_physical_expr(Arc::clone(value))?;
```